artiq/applets/plot_hist.py

```diff
@@ -24,10 +29,22 @@ def data_changed(self, data, mods, title):
             x = list(range(len(y)+1))
 
         if len(y) and len(x) == len(y) + 1:
+            self.flag = True
+            self.timer.stop()
             self.clear()
             self.plot(x, y, stepMode=True, fillLevel=0,
                       brush=(0, 0, 255, 150))
             self.setTitle(title)
+        else:
+            if self.flag:
+                self.flag = False
+                self.clear()
```

Comment:
Why clear here?

Reply:
If there is a successful plot already, simply add the `TextItem` without clear will make the text appear below the plot. Is that what we want?
![image](https://user-images.githubusercontent.com/43180603/125567620-b03ab11b-38a1-4c67-b6c7-3b6321664c23.png)

Reply:
What's more important, if another wrong dataset is submitted, the warning message will be rendered again and looks not good.

Reply:
What about clearing in ``length_warning`` then?

Reply:
ok